cmd/flakechecker/bq_client.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"cloud.google.com/go/bigquery"
+	"context"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/api/iterator"
+	"regexp"
+	"time"
+)
+
+const projectID = "acs-san-stackroxci"
+const queryTimeout = 1 * time.Minute
+
+type biqQueryClient interface {
+	GetRatioForTest(flakeTestConfig *flakeCheckerRecord, testName string) (int, int, error)
+}
+
+type biqQueryClientImpl struct {
+	client *bigquery.Client
+}
+
+func getNewBigQueryClient() (biqQueryClient, error) {
+	ctx := context.Background()
+
+	client, err := bigquery.NewClient(ctx, projectID)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating BigQuery client")
+	}
+
+	return &biqQueryClientImpl{client: client}, nil
+}
+
+func getFilteredTestName(name string) string {
+	// Apply the same filtering used in DB to generate test names not influenced with version Z number.
+	regFilterTestName := regexp.MustCompile(`(.*)((-v|: )\d{3}\.\d+\.)(\d+)(.*)`)
+	return regFilterTestName.ReplaceAllString(name, `${1}${2}z${5}`)
+}
+
+func (c *biqQueryClientImpl) GetRatioForTest(flakeTestConfig *flakeCheckerRecord, testName string) (int, int, error) {
+	query := c.client.Query(
+		`SELECT JobName, FilteredName, Classname, TotalAll, FailRatio
+    FROM acs-san-stackroxci.ci_metrics.stackrox_tests_recent_flaky_tests
+    WHERE JobName = @jobName AND FilteredName = @filteredName AND Classname = @classname
```

Comment:
How about better formatting for future changes?
```sql
SELECT 
  JobName, 
  FilteredName, 
  Classname, 
  TotalAll, 
  FailRatio 
FROM 
  "acs-san-stackroxci.ci_metrics.stackrox_tests_recent_flaky_tests"
WHERE 
  JobName = @jobName 
  AND FilteredName = @filteredName 
  AND Classname = @classname
```

Reply:
BTW we already have `acs-san-stackroxci` defined as const can we use it here? Maybe table should be passed as param?

Reply:
- formatting is applied
- table name will stay fixed, but it's improved with ` `` ` around it. Value `acs-san-stackroxci` has two different purposes. Constant is used as information on a project for authorization, and in another case (in SQL), it represents a project where data is stored. Tbh. This can be two different projects.
- we can not define table name as a param: From docs: `Parameters cannot be used as substitutes for identifiers, column names, table names, or other parts of the query.`